js-api-spec/node-package-importer.node.test.ts

```diff
@@ -0,0 +1,580 @@
+              },
+            }),
+          },
+        }));
+
+      it(`${key} with index`, () =>
+        testPackageImporter({
+          input: '@use "pkg:foo/subdir";',
+          output: 'a {b: c;}',
+          files: {
+            'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+            'node_modules/foo/src/sass/_other.scss': 'g {h: i}',
+            'node_modules/foo/src/sass/subdir/index.scss': 'a {b: c}',
+            'node_modules/foo/package.json': JSON.stringify({
+              exports: {
+                './subdir/index.scss': {
+                  [key]: './src/sass/subdir/index.scss',
+                },
+              },
+            }),
+          },
+        }));
+    });
+
+    it('compiles with first conditional match found', () =>
+      testPackageImporter({
+        input: '@use "pkg:foo";',
+        output: 'a {from: sassCondition;}',
+        files: {
+          'node_modules/foo/src/sass/_styles.scss': 'd {from: styleCondition}',
+          'node_modules/foo/src/sass/_other.scss': 'a {from: sassCondition}',
+          'node_modules/foo/package.json': JSON.stringify({
+            exports: {
+              '.': {
+                sass: './src/sass/_other.scss',
+                style: './src/sass/_styles.scss',
+              },
+            },
+          }),
+        },
+      }));
+
+    it('throws if multiple exported paths match', () =>
+      sandbox(dir => {
+        dir.write({
+          'node_modules/foo/src/sass/_styles.scss': 'd {e: f}',
+          'node_modules/foo/src/sass/_other.scss': 'a {b: c}',
+          'node_modules/foo/package.json': JSON.stringify({
+            exports: {
+              './index.scss': {sass: './src/sass/_other.scss'},
+              './_index.sass': {sass: './src/sass/_styles.scss'},
+            },
+          }),
+        });
+        dir.chdir(() => {
+          expect(() =>
+            compileString('@use "pkg:foo";', {
+              importers: [nodePackageImporter],
+            })
+          ).toThrowSassException({
+            includes: 'multiple potential resolutions',
+          });
+        });
+      }));
+
+    it('resolves string export', () =>
+      testPackageImporter({
+        input: '@use "pkg:foo";',
+        output: 'a {b: c;}',
+        files: {
+          'node_modules/foo/src/sass/_other.scss': 'a {b: c}',
+          'node_modules/foo/package.json': JSON.stringify({
+            exports: './src/sass/_other.scss',
+          }),
+        },
+      }));
+
+    describe('wildcards', () => {
+      it('resolves with partial', () =>
+        testPackageImporter({
+          input: '@use "pkg:foo/other";',
+          output: 'a {b: c;}',
+          files: {
+            'node_modules/foo/src/sass/_other.scss': 'a {b: c}',
+            'node_modules/foo/package.json': JSON.stringify({
+              exports: {'./*.scss': './src/sass/*.scss'},
+            }),
+          },
+        }));
+
+      it('resolves with full wildcard path', () =>
```

Comment:
I'd mention that this is about the combination of a wildcard _and_ a `sass:` key.